src/tools/rbd/action/Bench.cc

```diff
@@ -267,27 +305,23 @@ int do_bench(librbd::Image& image, io_type_t io_type,
   std::chrono::duration<double> last = std::chrono::duration<double>::zero();
   uint64_t ios = 0;
 
-  std::vector<uint64_t> thread_offset;
   uint64_t i;
-  uint64_t seq_chunk_length = (size / io_size / io_threads) * io_size;;
-
-  // disturb all thread's offset
-  for (i = 0; i < io_threads; i++) {
-    uint64_t start_pos = 0;
-    switch (io_pattern) {
-    case IO_PATTERN_RAND:
-      start_pos = (rand() % (size / io_size)) * io_size;
-      break;
-    case IO_PATTERN_SEQ:
-      start_pos = seq_chunk_length * i;
```

Comment:
Note strange behavior in old code when io_size * io_threads > size, seq_chunk_length becomes 0 and you do not get sequential I/O. This corner case is fixed below